instrumentation/httpclient-jdk11/src/main/java/jdk/internal/net/http/HttpClientImpl_Instrumentation.java

```diff
@@ -0,0 +1,51 @@
+package jdk.internal.net.http;
+
+import com.newrelic.api.agent.Segment;
+import com.newrelic.api.agent.Trace;
+import com.newrelic.api.agent.weaver.MatchType;
+import com.newrelic.api.agent.weaver.Weave;
+import com.newrelic.api.agent.weaver.Weaver;
+import com.nr.agent.instrumentation.httpclient.Util;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.concurrent.CompletableFuture;
+
+import static com.nr.agent.instrumentation.httpclient.Util.*;
+
+@Weave(originalName = "jdk.internal.net.http.HttpClientImpl", type = MatchType.ExactClass)
+final class HttpClientImpl_Instrumentation {
+
+    @Trace
+    public <T> HttpResponse<T> send(HttpRequest req, HttpResponse.BodyHandler<T> responseHandler)
+            throws IOException, InterruptedException {
+        URI uri = req.uri();
+        Segment segment = startSegment(uri);
+        HttpResponse<T> response;
```

Comment:
This block would be one of those opportunities we discussed where a private (maybe static) method might help document what it does real quick.  'tryToProcessResponse' perhaps is quicker to understand than scanning the code.  Just a thought, not meant to be a criticism by ANY means.

Reply:
Yeah, I think it would not work or be helpful here now that I look at more closely.  The `Weaver.callOriginal()` and segment stuff probably works better inline here rather than pulled out.  I'll try to give a better example at some point.